bentoml/_internal/frameworks/mlflow.py

```diff
@@ -0,0 +1,213 @@
+
+    Args:
+        bento_model: Either the tag of the model to get from the store, or a BentoML
+            ``~bentoml.Model`` instance to load the model from.
+
+    Returns:
+        The MLFlow model loaded as PyFuncModel from the BentoML model store.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+        pyfunc_model = bentoml.mlflow.load_model('my_model:latest')
+        pyfunc_model.predict( input_df )
+    """  # noqa
+    if not isinstance(bento_model, bentoml.Model):
+        bento_model = get(bento_model)
+
+    if bento_model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {bento_model.tag} was saved with module {bento_model.info.module}, not loading with {MODULE_NAME}."
+        )
+
+    return mlflow.pyfunc.load_model(bento_model.path_of(MLFLOW_MODEL_FOLDER))
+
+
+def import_model(
+    name: str,
+    model_uri: str,
+    *,
+    signatures: dict[str, ModelSignatureDict | ModelSignature] | None = None,
+    labels: dict[str, str] | None = None,
+    custom_objects: dict[str, t.Any] | None = None,
+    metadata: dict[str, t.Any] | None = None,
+    # ...
+) -> bentoml.Model:
+    """
+    Save a <FRAMEWORK> model instance to the BentoML model store.
+
+    Args:
+        name:
+            The name to give to the model in the BentoML store. This must be a valid
+            :obj:`~bentoml.Tag` name.
+        model_uri:
+            The <FRAMEWORK> model to be saved.
+        signatures:
+            Signatures of predict methods to be used. If not provided, the signatures
+            default to {"predict": {"batchable": False}}. See
+            :obj:`~bentoml.types.ModelSignature` for more details.
+        labels:
+            A default set of management labels to be associated with the model. For
+            example: ``{"training-set": "data-v1"}``.
+        custom_objects:
+            Custom objects to be saved with the model. An example is
+            ``{"my-normalizer": normalizer}``. Custom objects are serialized with
+            cloudpickle.
+        metadata:
+            Metadata to be associated with the model. An example is ``{"param_a": .2}``.
+
+            Metadata is intended for display in a model management UI and therefore all
+            values in metadata dictionary must be a primitive Python type, such as
+            ``str`` or ``int``.
+
+    Returns:
+        A :obj:`~bentoml.Model` instance referencing a saved model in the local BentoML
+        model store.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+
+        bentoml.mlflow.import_model(
+            'my_mlflow_model',
+            model_uri="runs:/<mlflow_run_id>/run-relative/path/to/model",
+            signatures={
+                "batchable": True
+            }
+        )
+    """
+    context = ModelContext(
+        framework_name="mlflow",
+        framework_versions={"mlflow": mlflow.__version__},
```

Comment:
we should use `get_pkg_version` here from `._internal.utils.pkg.get_pkg_version`

Reply:
import_metadata.version is nice for using distribution name instead of package name and does not require importing the library. but in the case of a library already imported, I'm not sure we have to use that API.  I will change it to keep it consistent for now. We can revisit the get_pkg_version implementation a bit later.

it is a bit tricky for developers since it should really be `get_package_version_by_distribution_name` - for example, you need get_pkg_version('scikit-learn') instead of get_pkg_version('sklearn')

Reply:
you are correct.